backend/src/main/kotlin/org/loculus/backend/utils/MetadataEntry.kt

```diff
@@ -29,42 +36,65 @@ fun findAndValidateSubmissionIdHeader(headerNames: List<String>): String {
 }
 
 fun metadataEntryStreamAsSequence(metadataInputStream: InputStream): Sequence<MetadataEntry> {
-    val csvParser = CSVFormat.TDF.builder().setHeader().setSkipHeaderRecord(true).get()
-        .parse(InputStreamReader(metadataInputStream))
+    val csvParser = try {
+        CSVFormat.TDF.builder().setHeader().setSkipHeaderRecord(true).get()
+            .parse(InputStreamReader(metadataInputStream))
+    } catch (e: CSVException) {
+        throw invalidTsvFormatException(e)
+    }
 
     val headerNames = csvParser.headerNames
     val submissionIdHeader = findAndValidateSubmissionIdHeader(headerNames)
 
-    return csvParser.asSequence().map { record ->
-        val submissionId = record[submissionIdHeader]
-        if (submissionId.isNullOrEmpty()) {
-            throw UnprocessableEntityException(
-                "A row in metadata file contains no $submissionIdHeader: $record",
-            )
-        }
-
-        if (submissionId.any { it.isWhitespace() }) {
-            throw UnprocessableEntityException(
-                "A value for $submissionIdHeader contains whitespace: $record",
-            )
-        }
-
-        val metadata = record.toMap().filterKeys { it != submissionIdHeader }
-        MetadataEntry(submissionId, metadata)
-    }.onEach { entry ->
-        if (entry.metadata.isEmpty()) {
-            throw UnprocessableEntityException(
-                "A row in metadata file contains no metadata columns: $entry",
-            )
+    return sequence {
+        try {
+            csvParser.asSequence().withIndex().forEach { (index, record) ->
+                val recordNumber = index + 1 // First data record is #1 (header not counted)
+                val submissionId = record[submissionIdHeader]
+                if (submissionId.isNullOrEmpty()) {
+                    val rowValues = record.toList().joinToString("', '", prefix = "['", postfix = "']")
+                    throw UnprocessableEntityException(
+                        "Record #$recordNumber in the metadata file contains no value for '$submissionIdHeader'. Row: $rowValues",
```

Comment:
Thanks for the changes - I expect this might often be too long? we could truncate? Not blocking we can iterate later if you prefer

Reply:
Very long error messages do not fit on the screen and the beginning (which has the important stuff) goes off the screen iirc